aws/cfn/bridge/resources.py

```diff
@@ -90,21 +82,69 @@ def __init__(self, name, source_file, options):
 
         # Determine our default action
-        action = options.get(_OPTION_DEFAULT_ACTION, None)
+        self._default_action = options.get(_OPTION_DEFAULT_ACTION, None)
+
+        config_provider = options.get(_OPTION_CONFIG_PROVIDER)
+        if config_provider:
+            self.override_config(config_provider)
+
+        if not self._queue_url:
+            raise ValueError(u"[%s] in '%s' is missing 'queue_url' attribute" % (name, source_file))
+
+        # Try to parse the region from the queues/sqs prefixed urls.
+        region_match = re.match(r"https?://(?:queue|sqs)\.([^.]+?)\.amazonaws\..+", self._queue_url, re.I | re.U)
+        if region_match:
+            self._region = region_match.group(1)
+
+        if not self._region:
+            raise ValueError(u"[%s] in '%s' must define 'region' attribute" % (name, source_file))
 
         # Set our actions for each type of event
-        self._create_action = options.get(_OPTION_CREATE_ACTION, action)
+        self._create_action = options.get(_OPTION_CREATE_ACTION, self._default_action)
         if not self._create_action:
             raise ValueError(u"[%s] in '%s' must define %s" % (name, source_file, _OPTION_CREATE_ACTION))
 
-        self._delete_action = options.get(_OPTION_DELETE_ACTION, action)
+        self._delete_action = options.get(_OPTION_DELETE_ACTION, self._default_action)
         if not self._delete_action:
             raise ValueError(u"[%s] in '%s' must define %s" % (name, source_file, _OPTION_DELETE_ACTION))
 
-        self._update_action = options.get(_OPTION_UPDATE_ACTION, action)
+        self._update_action = options.get(_OPTION_UPDATE_ACTION, self._default_action)
         if not self._update_action:
             raise ValueError(u"[%s] in '%s' must define %s" % (name, source_file, _OPTION_UPDATE_ACTION))
 
+    def override_config(self, config_provider):
+        providers = config_provider.rsplit('.', 2)
+        if len(providers) != 3:
+            raise ValueError(
+                'Provide <python lib path>.<class name>.<func_name> '
+                'ex) example.com.CfnBrdigeConfigProviderClass.get_option_func')
+
+        module_name = providers[0]
+        class_name = providers[1]
+        func_name = providers[2]
+
+        try:
+            module_provider = importlib.import_module(module_name)
+        except ImportError:
+            raise ValueError(u"Cloud not import %s" % module_name)
+
+        try:
+            class_provider = getattr(module_provider, class_name)
+        except AttributeError:
+            raise ValueError(u"Class %s is not defined" % class_name)
+
+        class_provider = class_provider()
+        try:
+            func_provider = getattr(class_provider, func_name)
+        except AttributeError:
+            raise ValueError(u"Func %s is not defiend" % func_name)
+
+        dict_options = func_provider()
+        for key, value in dict_options.iteritems():
```

Comment:
The provider may want to set values that are not set by default. Just pass in the current config and allow it to modify it as much as it needs.